clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,474 @@
+	// We assume 4 pods will be created per second.
+	podReadyTimeOutFactor = 0.25
+
+	// podReadyPollInterval is used in wait.poll(). Every 2seconds,check whether pods are ready.
+	podReadyPollInterval = 2 * time.Second
+
+	// initialDelay in seconds after which pod starts sending traffic.
+	// Delay is used to synchronize all client pods to send traffic at same time.
+	initialDelayForTestExecution = 5 * time.Second
+
+	// networkPerformanceMetricsName indicates the measurement name
+	networkPerformanceMetricsName = "NetworkPerformanceMetrics"
+
+	netperfNamespace = "netperf"
+)
+
+const (
+	networkTestRequestFilePath = manifestPathPrefix + "/" + "networktestrequests.yaml"
+	customResourceGroup        = "clusterloader.io"
+	customResourceKind         = "NetworkTestRequest"
+	customResourceVersion      = "v1alpha1"
+
+	crdManifestFilePath           = manifestPathPrefix + "/" + "*CustomResourceDefinition.yaml"
+	customResourceDefinitionGroup = "apiextensions.k8s.io"
+	customResourceDefinitionKind  = "CustomResourceDefinition"
+	customResourceDefinitionName  = "networktestrequests.clusterloader.io"
+
+	clusterRoleBindingFilePath = manifestPathPrefix + "/" + "roleBinding.yaml"
+	rbacGroup                  = "rbac.authorization.k8s.io"
+	rbacKind                   = "ClusterRoleBinding"
+	rbacName                   = "networktestrequests-rbac"
+)
+
+func init() {
+	klog.Info("Registering Network Performance Measurement")
+	if err := measurement.Register(networkPerformanceMetricsName, createNetworkPerformanceMeasurement); err != nil {
+		klog.Fatalf("Cannot register %s: %v", networkPerformanceMetricsName, err)
+	}
+}
+
+func createNetworkPerformanceMeasurement() measurement.Measurement {
+	return &networkPerformanceMeasurement{}
+}
+
+type networkPerformanceMeasurement struct {
+	k8sClient       kubernetes.Interface
+	framework       *framework.Framework
+	podReplicas     int
+	podRatio        string
+	numberOfServers int
+	numberOfClients int
+	podRatioType    string
+	testDuration    int
+	protocol        string
+	templateMapping map[string]interface{}
+	// workerPodList stores list of podData for every worker node.
+	workerPodList map[string][]workerPodData
+
+	// metricVal stores MetricResponse received from worker node.
+	metricVal  [][]float64
+	metricLock sync.Mutex
+
+	podPairNumber int
+
+	podPairCh chan uniquePodPair
+
+	// startTimeStampForTestExecution is a futureTimeStamp sent to all pods to start
+	// sending the traffic at the same time.
+	startTimeStampForTestExecution int64
+
+	resultSummary testResultSummary
+}
+
+func (npm *networkPerformanceMeasurement) Execute(config *measurement.Config) ([]measurement.Summary, error) {
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		klog.Error("error starting action: ", err)
+		return nil, err
+	}
+
+	switch action {
+	case "start":
+		if err = npm.validate(config); err != nil {
+			klog.Error("error in validation action: ", err)
```

Comment:
there is no need to log those errors - these will be logged by the framework anyway.
Please drop this logging (here and in other places of Execute method).

Reply:
Modified as per suggestion.